server/src/main/java/org/opensearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -159,7 +159,8 @@ protected NodeGatewayStartedShards nodeOperation(NodeRequest request) {
                 nodeEnv.availableShardPaths(request.shardId)
             );
             if (shardStateMetadata != null) {
-                if (indicesService.getShardOrNull(shardId) == null) {
+                if (indicesService.getShardOrNull(shardId) == null
+                    && shardStateMetadata.indexDataLocation == ShardStateMetadata.IndexDataLocation.LOCAL) {
```

Comment:
The reason I added a field to ShardStateMetadata was so that I could use that metadata on disk to avoid the `Store.tryOpenIndex` call below that assumes all indexes should have readable Lucene data on disk. If anyone has alternative suggestions here I'd love to hear it. Issue #3952 is another case where this assumption is causing problems. Unfortunately, I'm not sure that my solution here would be applicable to that issue.

Reply:
So what are the consequences of skipping the shards for REMOTE indices? The index shards stay as not strarted / not initialized?

Reply:
The check here is to determine if the shard on disk is "good". To do so, it assumes that the data path should have Lucene data that is readable with a `NIOFSDirectory`. That assumption is not valid for this new type of "remote" shard. So the change here results in saying the shard is "good" without attempting to read data that doesn't exist.

As issue #3952 demonstrates, this is sort of broken. We provide APIs to use a different `org.apache.lucene.store.Directory` implementation via IndexStorePlugin, but then this code assumes that there is data on disk readable with a NIOFSDirectory.